### PR TITLE
Dem benchmark use mainscript + multiprocess benchmarking

### PR DESCRIPTION
**Pull request by farrufat-cimne**

### CODE REVIEW

**Comment by farrufat-cimne** [comment]
log:

The complete list of benchmarks are included at the end of this message as a quick reference.

========== DEM BENCHMARKING RESULTS ==========

=========== DEM DISCONTINUUM TESTS ===========

==== TSUJI PAPER BENCHMARKS. SLIDING REGIME ==

DEM Benchmark 7: OK!........ Test 7 SUCCESSFUL
DEM Benchmark 1: OK!........ Test 1 SUCCESSFUL
DEM Benchmark 6: OK!........ Test 6 SUCCESSFUL
DEM Benchmark 4: OK!........ Test 4 SUCCESSFUL
DEM Benchmark 9: OK!........ Test 9 SUCCESSFUL


== BASIC CONTINUUM TESTS ==

DEM Benchmark 20: OK!........ Test 20 SUCCESSFUL
DEM Benchmark 2: OK!........ Test 2 SUCCESSFUL
DEM Benchmark 21: OK!........ Test 21 SUCCESSFUL
DEM Benchmark 8: OK!........ Test 8 SUCCESSFUL

==== THORNTON PAPER TESTS. FULL REGIME. HERTZIAN LAW ====

DEM Benchmark 11: OK!........ Test 11 (e=0.25) SUCCESSFUL
DEM Benchmark 24: OK!........ Test 24 SUCCESSFUL
DEM Benchmark 22: OK!........ Test 22 SUCCESSFUL
DEM Benchmark 23: OK!........ Test 23 SUCCESSFUL


===== DISCONTINUUM CLUSTERS TESTS =====

DEM Benchmark 30: OK!........ Test 30 SUCCESSFUL
DEM Benchmark 25: OK!........ Test 25 SUCCESSFUL
DEM Benchmark 31: OK!........ Test 31 SUCCESSFUL
DEM Benchmark 11: OK!........ Test 11 (e=0.50) SUCCESSFUL
DEM Benchmark 11: OK!........ Test 11 (e=0.75) SUCCESSFUL
DEM Benchmark 3: OK!........ Test 3 SUCCESSFUL
DEM Benchmark 11: OK!........ Test 11 (e=0.90) SUCCESSFUL

===== THORNTON PAPER TESTS. FULL REGIME. LINEAR LAW =====

DEM Benchmark 10: OK!........ Test 10 (e=0.25) SUCCESSFUL
DEM Benchmark 5: OK!........ Test 5 SUCCESSFUL
DEM Benchmark 10: OK!........ Test 10 (e=0.50) SUCCESSFUL
DEM Benchmark 10: OK!........ Test 10 (e=0.75) SUCCESSFUL
DEM Benchmark 10: OK!........ Test 10 (e=0.90) SUCCESSFUL
DEM Benchmark 33: OK!........ Test 33 SUCCESSFUL
DEM Benchmark 32: OK!........ Test 32 SUCCESSFUL

File: applications/DEM_application/python_scripts/DEM_procedures.py [hunk 1]
```diff
@@ -751,6 +751,10 @@ def CreateDirectories(self, main_path, problem_name, run_code = ''):
             if not os.path.isdir(directory):
                 os.makedirs(str(directory))
 
+            #if os.path.exists(directory):
+            #    shutil.rmtree(directory)
+            #os.makedirs(directory)
+
```
Comment (maceligueta) [inline applications/DEM_application/python_scripts/DEM_procedures.py]: can we remove these comments?

File: applications/DEM_application/test_examples/DEM_run_all_benchmarks.py [hunk 2]
```diff
@@ -75,43 +75,43 @@ def Run():
         try:
             if platform.system()=="Windows":
                 os.system("setenv OMP_NUM_THREADS 1") # Is that the correct way to run on Windows?
-                subprocess.check_call(["python", path + "/DEM_benchmarks.py", str(benchmark), ">", "BenchTemp.txt"], stdout=f, stderr=f)
+                subprocess.check_call(["python", path + "/DEM_benchmarks.py", str(benchmark), ">", "BenchTemp.info"], stdout=f, stderr=f)
                 os.system("setenv OMP_NUM_THREADS 16") # Trying to set a 'default' value
```
Comment (maceligueta) [inline applications/DEM_application/test_examples/DEM_run_all_benchmarks.py]: This default value is wrong. I propose resetting the variable (then the code uses the max number of threads). Let's take the chance to fix this now.

File: applications/DEM_application/test_examples/DEM_run_all_benchmarks.py [hunk 3]
```diff
@@ -75,43 +75,43 @@ def Run():
...
-                os.system("export OMP_NUM_THREADS=1")
+                os.environ['OMP_NUM_THREADS']='1'
                 if sys.version_info >= (3, 0):
-                    subprocess.check_call(["python3", path + "/DEM_benchmarks.py", str(benchmark), ">", "BenchTemp.txt"], stdout=f, stderr=f)
+                    subprocess.check_call(["python3", path + "/DEM_benchmarks.py", str(benchmark), ">", "BenchTemp.info"], stdout=f, stderr=f)
                     
                 else:
-                    subprocess.check_call(["python", "-3", path + "/DEM_benchmarks.py", str(benchmark), ">", "BenchTemp.txt"], stdout=f, stderr=f)
+                    subprocess.check_call(["python", "-3", path + "/DEM_benchmarks.py", str(benchmark), ">", "BenchTemp.info"], stdout=f, stderr=f)
                 os.system("export OMP_NUM_THREADS=16") # Trying to set a 'default' value
```
Comment (maceligueta) [inline applications/DEM_application/test_examples/DEM_run_all_benchmarks.py]: Same comment here.

**Review by maceligueta — CHANGES_REQUESTED** [review]
I left a couple of comments...

File: applications/DEM_application/test_examples/DEM_run_all_benchmarks.py (hunk 2 above)
Comment (farrufat-cimne) [reply applications/DEM_application/test_examples/DEM_run_all_benchmarks.py]: I guess OMP_DYNAMIC=true will do the job

**Comment by farrufat-cimne** [comment]
changed default to omp_num_threads=

**Review by maceligueta — APPROVED** [review]

**Comment by pooyan-dadvand** [comment]
I took this opportunity to ask if you can add some of the benchmarks to the [application cases](https://github.com/KratosMultiphysics/Kratos/wiki/Application-Cases) in wiki. It would be a great contribution!

**Pull request merged**